tuxemon/ai.py

```diff
@@ -112,10 +119,12 @@ def track_next_use(
         """
         # tracks next_use NPC ai
         filter_moves = []
-        # it choses among the last 4 moves
+        # it chooses among the last 4 moves
         for mov in monster.moves[-monster.max_moves :]:
             if mov.next_use <= 0:
-                filter_moves.append(mov)
+                # checks technique conditions
+                if mov.validate(self.opponent(opponents)):

```

Comment:
You use always check the first opponent here, but could end up returning a different one.
I would do something like this:

```
actions = []

for mov in monster.moves[-monster.max_moves :]:
    if mov.next_use <= 0:
    	for opponent in opponents:
    		if mov.validate(opponent):
	    	    actions.append((mov, opponent))
if not filter_moves:
    skip = Technique("skip")
    return skip, choice(opponents)
else:
    return choice(actions)
```
Please test it.

Reply:
It works without issues, I don't know why I didn't think about inserting the for before instead of creating a new def.